### PR TITLE
New copy_button style: mac

### PR DESCRIPTION
**Pull request by stevenjoezhang**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] [Docs](https://github.com/theme-next/theme-next.org/tree/source/source/docs) in [NexT website](https://theme-next.org/docs/) have been added / updated (for features).

## PR Type
**What kind of change does this PR introduce?**

- [ ] Bugfix.
- [x] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue resolved: N/A

## What is the new behavior?

- Screenshots with this changes: N/A
- Link to demo site with this changes: N/A

### How to use?
In NexT `_config.yml`:
```yml
...
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

**Comment by 1v9** [comment]
Suggestion: move `layout/_third-party/copy-code.swig` to `layout/path/to/copy-code.swig`, I don't think this is 3rd party.

**Comment by stevenjoezhang** [comment]
How about moving it to `util.js`?

**Comment by 1v9** [comment]
@stevenjoezhang Maybe `_scripts`? I'm not sure 😂

**Comment by 1v9** [comment]
Or we can create a new dir for features like this.

**Comment by stevenjoezhang** [comment]
Moved to `utils.js`, seems OK

File: _config.yml [hunk 1]
```diff
@@ -335,7 +335,7 @@ codeblock:
     enable: false
     # Show text copy result
     show_result: false
-    # Style: only `flat` is currently available, leave it blank if you prefer default theme
+    # Available values: default | flat | mac
```
Comment (1v9) [inline _config.yml]: is `default` useful?

File: source/css/_common/components/highlight/copy-code.styl [hunk 2]
```diff
@@ -1,47 +1,69 @@
...
+  position: absolute;
+  padding: 2px 6px;
   font-size: 13px;
   font-weight: 700;
   line-height: 20px;
   color: #333;
   white-space: nowrap;
   vertical-align: middle;
   cursor: pointer;
-  if hexo-config('codeblock.copy_button.style') == 'flat' {
```
Comment (1v9) [inline source/css/_common/components/highlight/copy-code.styl]: `===` is better?

Comment (stevenjoezhang) [reply source/css/_common/components/highlight/copy-code.styl]: Nope, there is no `===` operator in this language 😂

Comment (1v9) [reply source/css/_common/components/highlight/copy-code.styl]: Really? 🤨

Comment (stevenjoezhang) [reply source/css/_common/components/highlight/copy-code.styl]: Yes, you can try 🤔

File: _config.yml (hunk 1 above)
Comment (stevenjoezhang) [reply _config.yml]: No, default is blank

**Comment by jiangtj** [comment]
image
Seems not add `copycode.enable`?

**Comment by stevenjoezhang** [comment]
@jiangtj I thought it was added... fixed now

**Pull request merged**